x-pack/plugins/endpoint/public/applications/endpoint/store/routing/middleware.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AppAction } from '../action';
+import { MiddlewareFactory } from '../../types';
+import { flattenedPromise } from '../../lib/flattened_promise';
+
+export const routingMiddlewareFactory: MiddlewareFactory = (coreStart, history) => {
+  const [promise, resolve] = flattenedPromise();
+
+  return {
```

Comment:
@paul-tavares just to follow up with the comment @peluja1012 left on your pr, this is the new route action/middleware approach we're taking, would like to hear your thoughts